File: 1122-relative-sort-array/1122-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        map<int, int>mp;
        for(int i=0; i<arr1.size(); i++)
        {
            mp[arr1[i]]++;
        }
        
        vector<int>result;
        for(int i=0; i<arr2.size(); i++)
        {
            int size = mp[arr2[i]];
            for(int j=0; j<size; j++)
            {
                result.push_back(arr2[i]);
            }
        }
        
        sort(arr1.begin(), arr1.end());
        map<int,int>mp2;
        for(int i=0; i<arr2.size(); i++)
        {
            mp2[arr2[i]]++;
        }
        
        for(int i=0; i<arr1.size(); i++)
        {
            if(mp2.find(arr1[i]) == mp2.end())
            {
                result.push_back(arr1[i]);
            }
        }
        
        return result;
    }
};

File: 1122-relative-sort-array/1122-relative-sort-array.cs
public class Solution {
    public int[] RelativeSortArray(int[] arr1, int[] arr2) {
       Dictionary<int, int> mp = new Dictionary<int, int>();
        foreach (int num in arr1) {
            if (mp.ContainsKey(num)) {
                mp[num]++;
            } else {
                mp[num] = 1;
            }
        }

        // Create the result list based on the order in arr2
        List<int> result = new List<int>();
        foreach (int num in arr2) {
            if (mp.ContainsKey(num)) {
                for (int j = 0; j < mp[num]; j++) {
                    result.Add(num);
                }
                mp.Remove(num); // Remove to avoid adding duplicates later
            }
        }

        // Sort the remaining elements in arr1
        List<int> remainingElements = new List<int>();
        foreach (int num in arr1) {
            if (mp.ContainsKey(num)) {
                remainingElements.Add(num);
            }
        }
        remainingElements.Sort();

        // Add the remaining elements to the result list
        result.AddRange(remainingElements);

        // Convert the result list to an array and return it
        return result.ToArray();
    }
}